cmd/kubeadm/app/util/sentinelfile.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"os"
+	"io/ioutil"
+	"github.com/golang/glog"
+)
+
+var KubeadmSentinelFolder = "/var/lib/kubeadm"
+var KubeadmSentinelFile = KubeadmSentinelFolder + "/" + "kubeadm-sentinel"
+
+// write operation record to kubeadm-sentinel file when kubeadm init or kubeadm join
+func WriteSentineFile(content string) error {
+	// Writes a kubeadm-sentinel content down to disk  file with mode 0664
+
+	s := []byte(content)
+	if _, err := os.Stat(KubeadmSentinelFolder); err != nil {
+		err := os.MkdirAll(KubeadmSentinelFolder, 0664)
```

Comment:
make 0644 into a constant in this file:
`const kubeadmSentinelPermissions = 0664`